src/revlanguage/workspace/Environment.h

```diff
@@ -23,9 +23,19 @@ class RevObject;
      * @brief Environment: Base class for frames
      *
      * Instances of the Environment class are used to hold information about an
-     * evaluation or execution environment, such as the stack frame of a function call.
-     * Each environment has a pointer to the enclosing (parent) environment.
-     * A frame ('Environment') and its parents constitute an evaluation environment.
+     * evaluation or execution environment, such as the stack frame of a function call
+     * or a namespace.
+     *
+     * (a) Environment's that represent stack frames only define local variables and
+     *     contain a pointer to the enclosing (parent) environment to look up external variables.
+     *     Such environments contains a reference to the parent environment so that the parent
+     *     environment will not be destroyed before the stack frame environment.
+     *
+     * (b) Environment's that represent a namespace do not contain a pointer to the parent
+     *     Environment.  Instead, the parent environment holds a references to the namespace.
+     *     This ensures that the namespace environment will not be destroyed before the
+     *     enclosing environment.
```

Comment:
Some typos here... "Environment's" should read "Environments" (lines 29, 34), "environments contains" should read "environments contain" (line 31), the word "Environment" seems to be randomly capitalized (line 35), and "a references" should read "a reference" (line 35).

That said, thank you for explaining the logic behind this! I feel like the finer nuances of Rev scoping always escaped me, so documenting how it's implemented helps a lot.

Reply:
I fixed some typos.

You can read "Environment" as "an instance of the Environment class" and "environment" as "a mapping from names to variables".

Reply:
Thanks! At the risk of being super annoying, the phrase "a references" still occurs on line 35. I believe it should be singular (one reference per namespace).

Reply:
No worries, that's easy enough to fix.

Reply:
Fixed.